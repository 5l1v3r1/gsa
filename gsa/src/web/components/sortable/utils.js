/* Greenbone Security Assistant
 *
 * Authors:
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import uuid from 'uuid/v4';

import {createRow} from 'gmp/commands/dashboards';

export const createItem = (props, uuidFunc = uuid) => {
  const id = uuidFunc();

  return {
    id,
    ...props,
  };
};

export const removeItem = (rows, itemId) => rows.map(row => ({
  ...row,
  items: row.items.filter(item => item.id !== itemId),
})).filter(row => row.items.length > 0);

export const updateRow = (row, data) => ({
  ...row,
  ...data,
});

export const convertDefaultContent = (defaultContent = [], uuidFunc = uuid) => {
  return {
    rows: defaultContent.map(row => createRow(
      row.map(item => createItem({name: item}, uuidFunc)), undefined, uuidFunc
    )),
  };
};

// vim: set ts=2 sw=2 tw=80:
