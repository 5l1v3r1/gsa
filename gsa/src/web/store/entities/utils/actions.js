/* Greenbone Security Assistant
 *
 * Authors:
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */

export const types = {
  ENTITIES_LOADING_REQUEST: 'ENTITIES_LOADING_REQUEST',
  ENTITIES_LOADING_SUCCESS: 'ENTITIES_LOADING_SUCCESS',
  ENTITIES_LOADING_ERROR: 'ENTITIES_LOADING_ERROR',
};

export const createEntitiesActionCreators = entityType => ({
  request: filter => ({
    type: types.ENTITIES_LOADING_REQUEST,
    entityType,
    filter,
  }),
  success: (data, filter) => ({
    type: types.ENTITIES_LOADING_SUCCESS,
    entityType,
    filter,
    data,
  }),
  error: (error, filter) => ({
    type: types.ENTITIES_LOADING_ERROR,
    entityType,
    filter,
    error,
  }),
});

export const createLoadAllFunc = ({
  selector,
  actionCreators,
  entityType,
}) => ({gmp, filter, ...props}) => (dispatch, getState) => {
    const rootState = getState();
    const state = selector(rootState);

    if (state.isLoadingEntities(filter)) {
      // we are already loading data
      return Promise.resolve();
    }

    dispatch(actionCreators.request(filter));

    return gmp[entityType].getAll({filter}).then(
      response => dispatch(actionCreators.success(response.data, filter)),
      error => dispatch(actionCreators.error(error, filter)),
    );
  };

// vim: set ts=2 sw=2 tw=80:
