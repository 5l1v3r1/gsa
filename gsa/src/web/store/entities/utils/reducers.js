/* Greenbone Security Assistant
 *
 * Authors:
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import {combineReducers} from 'redux';

import {is_defined} from 'gmp/utils/identity';

export const filterIdentifier = filter => is_defined(filter) ?
  `filter:${filter.toFilterString()}` :
  'default';

export const createReducer = types => {

  const isLoading = (state = false, action) => {
    switch (action.type) {
      case types.REQUEST:
        return true;
      case types.SUCCESS:
      case types.ERROR:
        return false;
      default:
        return state;
    }
  };

  const error = (state = null, action) => {
    switch (action.type) {
      case types.SUCCESS:
        return null;
      case types.ERROR:
        return action.error;
      default:
        return state;
    }
  };

  const entities = (state = [], action) => {
    switch (action.type) {
      case types.SUCCESS:
        const {data = []} = action;
        return data.map(entity => entity.id);
      default:
        return state;
    }
  };

  const byId = (state = {}, action) => {
    switch (action.type) {
      case types.SUCCESS:
        const {data = []} = action;
        const nextState = {
          ...state,
        };
        data.forEach(d => nextState[d.id] = d);
        return nextState;
      default:
        return state;
    }
  };

  const combinedReducer = combineReducers({
    isLoading,
    error,
    entities,
  });

  return (state = {}, action) => {
    switch (action.type) {
      case types.REQUEST:
      case types.SUCCESS:
      case types.ERROR:
        const filterString = filterIdentifier(action.filter);
        return {
          ...state,
          byId: byId(state.byId, action),
          [filterString]: combinedReducer(state[filterString], action),
        };
      default:
        return state;
    }
  };
};

// vim: set ts=2 sw=2 tw=80:
