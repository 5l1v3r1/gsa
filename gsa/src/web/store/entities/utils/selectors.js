/* Greenbone Security Assistant
 *
 * Authors:
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import {is_defined, has_value} from 'gmp/utils/identity';

class EntitiesSelector {

  constructor(state = {}) {
    this.state = state;
  }

  _getByFilter(filter) {
    const filterString = is_defined(filter) ? filter.toFilterString() :
      'default';
    return this.state[filterString];
  }

  getIsLoading(filter) {
    const state = this._getByFilter(filter);
    return is_defined(state) ? state.isLoading : false;
  }

  getError(filter) {
    const state = this._getByFilter(filter);
    return is_defined(state) ? state.error : undefined;
  }

  getEntities(filter) {
    const state = this._getByFilter(filter);
    if (is_defined(state) && has_value(state.entities)) {
      return state.entities.map(id => this.getEntity(id)).filter(is_defined);
    }
    return undefined;
  }

  getEntity(id) {
    return this.state.byId[id];
  }
};

export const createSelector = name => rootState =>
  new EntitiesSelector(rootState.entities[name]);

// vim: set ts=2 sw=2 tw=80:
