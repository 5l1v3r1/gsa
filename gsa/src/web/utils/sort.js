/* Greenbone Security Assistant
 *
 * Authors:
 * Timo Pollmeier <timo.pollmeier@greenbone.net>
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2017 - 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import logger from 'gmp/log';

import {parseInt, parseFloat} from 'gmp/parser';

import {isDefined, isFunction} from 'gmp/utils/identity';

const log = logger.getLogger('web.utils.sort');

const genericCompareAsc = (a, b) => {
  if (a > b) {
    return +1;
  }
  else if (b > a) {
    return -1;
  }
  return 0;
};

const genericCompareDesc = (a, b) => {
  if (a < b) {
    return +1;
  }
  else if (b < a) {
    return -1;
  }
  return 0;
};

const getProperty = (object, property) => {
  try {
    if (isFunction(property)) {
      return property(object);
    }

    return object[property];
  }
  catch (err) {
    log.error('Could not get property', property, object);
    return undefined;
  }
};

export const ipToNumber = original => {
  if (!isDefined(original)) {
    return undefined;
  }

  const split = original.split('.');
  if (split.length === 4) { // should be an ipv4 address
    let ret = 0;
    for (const item of split) {
      ret = ret << 8; // eslint-disable-line no-bitwise
      const number = parseInt(item);

      if (!isDefined(number)) { // wasn't a number. it's not an ip
        return original;
      }

      ret = ret | number; // eslint-disable-line no-bitwise
    }
    return ret;
  }

  // TODO support ipv6

  return original; // use original value for comparison
};

const getValue = (convertFunc, value, property, undefinedVal) => {
  const val = convertFunc(getProperty(value, property));

  return isDefined(val) ? val : undefinedVal;
};

const makeCompare = convertFunc => (property, undefinedVal) => reverse => {
  const valCompare = reverse ? genericCompareDesc : genericCompareAsc;

  return (a, b) => valCompare(
    getValue(convertFunc, a, property, undefinedVal),
    getValue(convertFunc, b, property, undefinedVal),
  );
};

export const makeCompareString = makeCompare(value => '' + value);

export const makeCompareNumber = makeCompare(parseFloat);

export const makeCompareDate = makeCompare(value => value);

export const makeCompareIp = makeCompare(ipToNumber);

export const makeCompareSeverity = (name = 'severity') =>
  makeCompareNumber(name, 0);

// vim: set ts=2 sw=2 tw=80:
