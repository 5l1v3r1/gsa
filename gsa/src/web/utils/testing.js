/* Copyright (C) 2018 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import 'jest-styled-components';

import React from 'react';

import {render as reactTestingRender} from 'react-testing-library';

import {Router} from 'react-router-dom';

import {Provider} from 'react-redux';

import {createMemoryHistory} from 'history';

import {
  toHaveAttribute,
  toHaveTextContent,
} from 'jest-dom';

import {hasValue} from 'gmp/utils/identity';

import configureStore from 'web/store';

expect.extend({
  toHaveAttribute,
  toHaveTextContent,
});

export * from 'react-testing-library';

export const render = ui => {
  const {container, ...other} = reactTestingRender(ui);
  return {
    container,
    element: hasValue(container) ? container.firstChild : undefined,
    ...other,
  };
};

export const rendererWithRouter = (
  history = createMemoryHistory({initialEntries: ['/']}),
) => ({
  render: ui => render(
    <Router history={history}>{ui}</Router>
  ),
  history,
});

export const rendererWithStore = (store = configureStore()) => ({
  render: ui => render(
    <Provider store={store}>{ui}</Provider>
  ),
  store,
});

export const rendererWithStoreAndRouter = (
  store = configureStore(),
  history = createMemoryHistory({initialEntries: ['/']}),
) => ({
  store,
  history,
  render: ui => render(
    <Provider store={store}>
      <Router history={history}>
        {ui}
      </Router>
    </Provider>
  ),
});

// vim: set ts=2 sw=2 tw=80:
