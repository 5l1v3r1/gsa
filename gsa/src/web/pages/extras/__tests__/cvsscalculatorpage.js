/* Copyright (C) 2019 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */

import React from 'react';
import {act} from 'react-dom/test-utils';

import {fireEvent, rendererWith, waitForElement} from 'web/utils/testing';

import CvssCalculator from 'web/pages/extras/cvsscalculatorpage';

const calculateScoreFromVector = jest.fn().mockReturnValue(
  Promise.resolve({
    data: 7.5,
  }),
);

const gmp = {
  cvsscalculator: {
    calculateScoreFromVector,
  },
  settings: {
    manualUrl: 'http://docs.greenbone.net/GSM-Manual/gos-5/',
  },
  user: {
    renewSession: jest.fn().mockReturnValue(
      Promise.resolve({
        data: 'foo',
      }),
    ),
  },
};

const onInteraction = jest.fn();

const location = {
  query: {cvssVector: 'AV:N/AC:L/Au:N/C:P/I:P/A:P'},
};

describe('CvssCalculator page tests', () => {
  test('Should render with default values', () => {
    const {render} = rendererWith({
      gmp,
      onInteraction,
      store: true,
    });

    const {element, getAllByTestId} = render(<CvssCalculator />);

    expect(element).toMatchSnapshot();
    const input = getAllByTestId('select-selected-value');
    waitForElement(() => element.querySelectorAll('input'));
    const vector = element.querySelectorAll('input');

    act(() => {
      expect(input[0]).toHaveTextContent('Local');
      expect(input[1]).toHaveTextContent('Low');
      expect(input[2]).toHaveTextContent('None');
      expect(input[3]).toHaveTextContent('None');
      expect(input[4]).toHaveTextContent('None');
      expect(input[5]).toHaveTextContent('None');

      expect(vector[0]).toHaveAttribute('value', 'AV:L/AC:L/Au:N/C:N/I:N/A:N');
    });
  });
  test('Should render userVector from url', () => {
    const {render} = rendererWith({
      gmp,
      onInteraction,
      store: true,
    });

    const {element, getAllByTestId} = render(
      <CvssCalculator location={location} />,
    );

    const input = getAllByTestId('select-selected-value');
    const vector = element.querySelectorAll('input');

    act(() => {
      expect(input[0]).toHaveTextContent('Network');
      expect(input[1]).toHaveTextContent('Low');
      expect(input[2]).toHaveTextContent('None');
      expect(input[3]).toHaveTextContent('Partial');
      expect(input[4]).toHaveTextContent('Partial');
      expect(input[5]).toHaveTextContent('Partial');

      expect(vector[0]).toHaveAttribute('value', 'AV:N/AC:L/Au:N/C:P/I:P/A:P');
    });
  });
  test('Changing userVector should change displayed select values', () => {
    const {render} = rendererWith({
      gmp,
      onInteraction,
      store: true,
    });

    const {element, getAllByTestId} = render(
      <CvssCalculator location={location} />,
    );

    const vector = element.querySelectorAll('input');

    fireEvent.change(vector[0], {
      target: {value: 'AV:N/AC:L/Au:N/C:N/I:P/A:P'},
    });

    const input = getAllByTestId('select-selected-value');

    act(() => {
      expect(input[0]).toHaveTextContent('Network');
      expect(input[1]).toHaveTextContent('Low');
      expect(input[2]).toHaveTextContent('None');
      expect(input[3]).toHaveTextContent('None');
      expect(input[4]).toHaveTextContent('Partial');
      expect(input[5]).toHaveTextContent('Partial');

      expect(vector[0]).toHaveAttribute('value', 'AV:N/AC:L/Au:N/C:N/I:P/A:P');
    });
  });
  test('Changing displayed select values should change userVector', async () => {
    const {render} = rendererWith({
      gmp,
      onInteraction,
      store: true,
    });

    const {element, getAllByTestId} = render(
      <CvssCalculator location={location} />,
    );

    const vector = element.querySelectorAll('input');

    const input = getAllByTestId('select-selected-value');

    act(() => {
      expect(input[0]).toHaveTextContent('Network');
      expect(input[1]).toHaveTextContent('Low');
      expect(input[2]).toHaveTextContent('None');
      expect(input[3]).toHaveTextContent('Partial');
      expect(input[4]).toHaveTextContent('Partial');
      expect(input[5]).toHaveTextContent('Partial');

      expect(vector[0]).toHaveAttribute('value', 'AV:N/AC:L/Au:N/C:P/I:P/A:P');
    });

    act(() => {
      const selectFields = getAllByTestId('select-open-button');

      fireEvent.click(selectFields[0]);
    });

    act(() => {
      const selectItems = getAllByTestId('select-item');

      fireEvent.click(selectItems[0]);
    });

    expect(vector[0]).toHaveAttribute('value', 'AV:L/AC:L/Au:N/C:P/I:P/A:P');
  });
});
