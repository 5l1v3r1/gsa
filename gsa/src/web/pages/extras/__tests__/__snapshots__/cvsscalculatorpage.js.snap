// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvssCalculator page tests Should render with default values 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c1 * {
  height: inherit;
  width: inherit;
}

.c8 {
  height: 50px;
  width: 50px;
  line-height: 50px;
}

.c8 * {
  height: inherit;
  width: inherit;
}

.c2 {
  margin: 10px 0px;
  padding-bottom: 1px;
  border-bottom: 2px solid #c8d3d9;
  position: relative;
}

.c5 {
  margin: 0 0 1px 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c6 {
  margin-right: 5px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding-bottom: 10px;
}

.c11 {
  display: inline-block;
  max-width: 100%;
  font-weight: bold;
  text-align: right;
  padding-left: 10px;
  padding-right: 10px;
  width: 16.66666667%;
  margin-left: 0;
}

.c12 {
  width: 83.33333333%;
  padding-left: 10px;
  padding-right: 10px;
}

.c19 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  margin: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c18 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c18 * {
  height: inherit;
  width: inherit;
}

.c15 {
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 1px 5px;
  background-color: #fff;
  color: #000;
  font-weight: normal;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 180px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  word-break: keep-all;
  white-space: nowrap;
  overflow: hidden;
  cursor: pointer;
}

.c22 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  display: block;
  height: 22px;
  color: #393637;
  background-color: #fff;
  background-image: none;
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  padding: 1px 8px;
}

.c22:-webkit-autofill {
  box-shadow: 0 0 0 1000px white inset;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c23 {
  height: 13px;
  box-sizing: content-box;
  display: inline-block;
  width: 100px;
  background: #393637;
  vertical-align: middle;
  text-align: center;
}

.c25 {
  z-index: 1;
  font-weight: bold;
  color: #fff;
  font-size: 9px;
  margin: 0;
  position: relative;
  top: -13px;
  padding-top: 1px;
}

.c24 {
  height: 13px;
  width: 0%;
  background: gray;
}

.c20 {
  width: 180px;
}

@media print {
  .c23 {
    background: none;
    border: 0;
  }
}

@media print {
  .c25 {
    color: black;
  }
}

@media print {
  .c24 {
    background: none;
  }
}

<div
  class="c0"
>
  <a
    href="http://docs.greenbone.net/GSM-Manual/gos-5/en/managing-secinfo.html#cvss"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="c1 "
      title="Help: CVSS Base Score Calculator"
    >
      <svg>
        help.svg
      </svg>
    </span>
  </a>
  <section>
    <div
      class="section-header c2 c3"
    >
      <h2
        class="c4 c5"
      >
        <div
          class="c6 c7"
        >
          <span
            class="c8 "
          >
            <svg>
              cvss_calculator.svg
            </svg>
          </span>
        </div>
        <div
          class="c7"
        >
          CVSSv2 Base Score Calculator
        </div>
      </h2>
      <div
        class="c9"
      />
    </div>
  </section>
  <h3>
    From Metrics
    :
  </h3>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Access Vector
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="Local"
          >
            Local
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Access Complexity
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="Low"
          >
            Low
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Authentication
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="None"
          >
            None
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Confidentiality
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="None"
          >
            None
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Integrity
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="None"
          >
            None
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Availability
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c14"
        width="180px"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="open menu"
          class="c15"
          data-toggle="true"
          role="button"
          type="button"
        >
          <div
            class="c16"
            data-testid="select-selected-value"
            title="None"
          >
            None
          </div>
          <div
            class="c17"
          >
            <span
              class="c18 c19"
              data-testid="select-open-button"
            >
              ▼
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h3>
    From Vector
    :
  </h3>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Vector
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <input
        class="c20 c21 c22"
        name="userVector"
        type="text"
        value="AV:L/AC:L/Au:N/C:N/I:N/A:N"
      />
    </div>
  </div>
  <h3>
    Results
    :
  </h3>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      CVSS Base Vector
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <span>
        AV:L/AC:L/Au:N/C:N/I:N/A:N
      </span>
    </div>
  </div>
  <div
    class="c10"
  >
    <label
      class="c11"
      data-testid="formgroup-title"
    >
      Severity
    </label>
    <div
      class="c12 c13"
      data-testid="formgroup-content"
      size="10"
    >
      <div
        class="c23"
        data-testid="progressbar-box"
        title="Log"
      >
        <div
          class="c24"
          data-testid="progress"
        />
        <div
          class="c25"
        >
          0.0 (Log)
        </div>
      </div>
    </div>
  </div>
</div>
`;
